src/openfl/display/DisplayObject.hx

```diff
@@ -1966,4 +2027,64 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 	}
 	
 	
+}
+
+@:access(openfl.display.DisplayObject)
+@:access(openfl.geom.ColorTransform)
+class RenderState {
+	
+	
+		private var __displayObject:DisplayObject;
+		private var __renderable:Bool;
+		private var __renderTransform:Matrix;
+		private var __worldAlpha:Float;
+		private var __worldBlendMode:BlendMode;
+		private var __worldColorTransform:ColorTransform;
+		private var __worldTrasform:Matrix;
+		
+		public function new (displayObject:DisplayObject) {
+			__displayObject = displayObject;
+		}
+	
+		public function save ():Void {
+			
+			if (__worldTrasform == null) {
+				
+				__worldTrasform = new Matrix ();
+				
+			}
+			__worldTrasform.copyFrom (__displayObject.__worldTransform);
+			
+			if (__renderTransform == null) {
+				
+				__renderTransform = new Matrix ();
+				
+			}
+			__renderTransform.copyFrom (__displayObject.__renderTransform);
+			
+			if (__worldColorTransform == null) {
+				
+				__worldColorTransform = new ColorTransform ();
+				
+			} 
+			__worldColorTransform.__copyFrom (__displayObject.__worldColorTransform);
+			
+			__worldAlpha = __displayObject.__worldAlpha;
+			__worldBlendMode = __displayObject.__worldBlendMode;
+			__renderable = __displayObject.__renderable;
+			
+		}
+
+		public function restore ():Void {
+			
+			__displayObject.__worldColorTransform.__copyFrom (__worldColorTransform);
```

Comment:
Minor: Keep the order from above, meaning `__worldTransform`, `__renderTransform` and `__worldColorTransform`?